### PR TITLE
Ekco version 0.12.0

### PR DESCRIPTION
**Pull request by areed**
Add ekco v0.12.0 with support for overrriding images in any pod created through the Kubernetes API.

### CODE REVIEW

File: addons/ekco/0.12.0/install.sh [hunk 1]
```diff
@@ -0,0 +1,362 @@
...
+        ekco_install_upgrade_service "$src"
+    else
+        ekco_remove_upgrade_service
+    fi
+
+    ekco_install_purge_node_command "$src"
+
+    # Wait for the pod image override mutating webhook to be created so when other add-ons are
+    # installed they will get their overridden image
+    if [ -n "$EKCO_POD_IMAGE_OVERRIDES" ]; then
```
Comment (areed) [inline addons/ekco/0.12.0/install.sh:60]: This block is new

File: addons/ekco/0.12.0/tmpl-configmap.yaml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    registry_cert_secret: registry-pki
+    kurl_proxy_cert_namespace: default
+    kurl_proxy_cert_secret: kotsadm-tls
+    kotsadm_kubelet_cert_namespace: default
+    kotsadm_kubelet_cert_secret: kubelet-client-cert
+    rook_priority_class: \"$EKCO_ROOK_PRIORITY_CLASS\"
+    host_task_image: replicated/ekco:v$EKCO_VERSION
+    host_task_namespace: kurl
+    enable_internal_load_balancer: $EKCO_ENABLE_INTERNAL_LOAD_BALANCER_BOOL
+    pod_image_overrides: \"$EKCO_POD_IMAGE_OVERRIDES\"
```
Comment (areed) [inline addons/ekco/0.12.0/tmpl-configmap.yaml:36]: this line is new

Comment (laverya) [reply addons/ekco/0.12.0/tmpl-configmap.yaml:36]: This is an array - does this method of escaping handle multiple elements?

File: kurlkinds/pkg/apis/cluster/v1beta1/installer_types.go [hunk 3]
```diff
@@ -226,17 +226,18 @@ type FirewalldConfig struct {
...
+	NodeUnreachableToleration   string   `json:"nodeUnreachableToleration,omitempty" yaml:"nodeUnreachableToleration,omitempty"`
+	RookShouldUseAllNodes       bool     `json:"rookShouldUseAllNodes,omitempty" yaml:"rookShouldUseAllNodes,omitempty"`
+	S3Override                  string   `json:"s3Override,omitempty" yaml:"s3Override,omitempty"`
+	ShouldDisableRebootServices bool     `json:"shouldDisableRebootServices,omitempty" yaml:"shouldDisableRebootServices,omitempty"`
+	ShouldDisableClearNodes     bool     `json:"shouldDisableClearNodes,omitempty" yaml:"shouldDisableClearNodes,omitempty"`
+	ShouldEnablePurgeNodes      bool     `json:"shouldEnablePurgeNodes,omitempty" yaml:"shouldEnablePurgeNodes,omitempty"`
+	Version                     string   `json:"version" yaml:"version"`
+	AutoUpgradeSchedule         string   `json:"autoUpgradeSchedule,omitempty" yaml:"autoUpgradeSchedule,omitempty"`
+	EnableInternalLoadBalancer  bool     `json:"enableInternalLoadBalancer,omitempty" yaml:"enableInternalLoadBalancer,omitempty"`
+	PodImageOverrides           []string `json:"podImageOverrides,omitempty" yaml:"podImageOverrides,omitempty"`
```
Comment (laverya) [inline kurlkinds/pkg/apis/cluster/v1beta1/installer_types.go:240]: should this be an array, or just a string?

File: web/src/installers/index.ts [hunk 4]
```diff
@@ -372,6 +372,7 @@ export interface EkcoConfig {
   shouldDisableClearNodes?: boolean;
   shouldEnablePurgeNodes?: boolean;
   rookShouldUseAllNodes?: boolean;
+  podImageOverrides?: Array<string>;
```
Comment (laverya) [inline web/src/installers/index.ts:375]: our first array!

**Review by laverya — APPROVED** [review]

**Pull request merged**